subsys/fs/nvs/nvs.c

```diff
@@ -737,7 +737,7 @@ ssize_t nvs_write(struct nvs_fs *fs, u16_t id, const void *data, size_t len)
 	size_t ate_size, data_size;
 	struct nvs_ate wlk_ate;
 	u32_t wlk_addr, rd_addr;
-	u16_t sector_freespace;
+	s32_t sector_freespace;
```

Comment:
No need to change this, a sector can maximum be 32kB.

Reply:
why? address space for each sector seems to be in the following range:
0x0000 - 0xFFFF

Reply:
If sector is filled up (delete ATE is written at the last possible position) free-space is below zero.

Reply:
@nvlsianpu, this should never be the case, it would appear when a delete is written to a location that would not allow space for an empty ate to be added. This cannot be allowed.